### PR TITLE
Non-payable functions revert if tokens attached

### PR DESCRIPTION
**Pull request by kpob**
* proxy caller ensures if the cargo purse is empty
* update OdraVm's call_contract to revert on error

### CODE REVIEW

File: odra-casper/proxy-caller/bin/proxy_caller.rs [hunk 1]
```diff
@@ -7,16 +7,20 @@ compile_error!("This binary only supports wasm32 target architecture!");
...
 fn call() {
     let proxy_call = ProxyCall::load_from_args();
-    let _: () = call_versioned_contract(
+    let _: () = runtime::call_versioned_contract(
         proxy_call.contract_package_hash,
         None,
         proxy_call.entry_point_name.as_str(),
         proxy_call.runtime_args
     );
+    if proxy_call.attached_value > 0.into() {
```
Comment (zie1ony) [inline odra-casper/proxy-caller/bin/proxy_caller.rs]:
```suggestion
    if !proxy_call.attached_value.is_zero() {
```

File: odra-casper/proxy-caller/bin/proxy_caller_with_return.rs [hunk 2]
```diff
@@ -20,5 +22,8 @@ fn call() {
         proxy_call.entry_point_name.as_str(),
         proxy_call.runtime_args
     );
+    if proxy_call.attached_value > 0.into() {
```
Comment (zie1ony) [inline odra-casper/proxy-caller/bin/proxy_caller_with_return.rs]: same here

File: odra-casper/proxy-caller/bin/proxy_caller.rs (hunk 1 above)
Comment (zie1ony) [inline odra-casper/proxy-caller/bin/proxy_caller.rs]: You have this code duplicated. It could live under `proxy_call.post_call_pruse_check()`.

**Review by zie1ony — APPROVED** [review]
Took me a bit to understand this small change. I think it is correct one.

**Review by kubaplas — APPROVED** [review]

**Pull request merged**